pallets/msa/src/lib.rs

```diff
@@ -709,27 +711,27 @@ impl<T: Config> Pallet<T> {
 	/// # Returns
 	/// * [`MessageSourceId`]
 	pub fn get_owner_of(key: &T::AccountId) -> Option<MessageSourceId> {
-		Self::get_key_info(&key).map(|info| info.msa_id)
+		Self::get_key_info(&key)
 	}
 
 	/// Fetches all the keys associated with a message Source Account
 	/// NOTE: This should only be called from RPC due to heavy database reads
 	pub fn fetch_msa_keys(msa_id: MessageSourceId) -> Vec<KeyInfoResponse<T::AccountId>> {
 		let mut response = Vec::new();
 		for key in Self::get_msa_keys(msa_id) {
-			if let Ok(info) = Self::try_get_key_info(&key) {
-				response.push(info.map_to_response(key));
+			if let Ok(_info) = Self::try_get_msa_from_account_id(&key) {
+				response.push(KeyInfoResponse { key, msa_id, nonce: 0 });
```

Comment:
I don't think this should have a nonce anymore.
```suggestion
				response.push(KeyInfoResponse { key, msa_id });
```